Marlin/src/feature/binary_stream.h

```diff
@@ -39,7 +39,7 @@ inline int bs_read_serial(const serial_index_t index) {
 
 #if ENABLED(BINARY_STREAM_COMPRESSION)
   static heatshrink_decoder hsd;
-  #if BOTH(ARDUINO_ARCH_STM32F1, SDIO_SUPPORT)
+  #if ENABLED(SDIO_SUPPORT) && EITHER(ARDUINO_ARCH_STM32F1, STM32F1)
```

Comment:
The define `ARDUINO_ARCH_STM32F1` would seem to only apply to `STM32F103VE_GTM32` which uses the Maple library. But, that environment is not assigned to any of the pins files in `pins.h`. Of course, `STM32F1` applies to all boards that use `HAL/STM32`.

Do we expect `ARDUINO_ARCH_STM32F1` to also be defined by some variants in the platform?

Or, more relevant to this change: Is there any reason not to align to a `size_t` boundary for _all_ targets?

Reply:
>Or, more relevant to this change: Is there any reason not to align to a size_t boundary for all targets?

I don't see a reason it needs to be conditional, would probably be better if the alignment was guaranteed on all platforms.

Reply:
Good question. but  a but  over my experience level, lets walk through it..

This is only needed for DMA with SDIO, at present that code is only in the one HAL STM32 
 ie Sd2Card_sdio_stm32duino.cpp

So it is definitely not needed for all targets, but would it harm them? I don't think so. But it could waste a few byes aligning when  it is not needed.  (is the compiler clever enough to put something else in those bytes vs waste them? I don't know) 

I would expect stm32f4 to also need this,  but my only f4 boards doesn't support SDIO.. so cannot test this.

Potentially dropping the  EITHER(ARDUINO_ARCH_STM32F1, STM32F1)  and only activating when SDIO_SUPPORT is used seems like a valid alternative with the information at hand.

Reply:
> Potentially dropping the EITHER(ARDUINO_ARCH_STM32F1, STM32F1) and only activating when SDIO_SUPPORT is used seems like a valid alternative with the information at hand.

This is the fix!

Reply:
Aligning might be helpful for all kinds of MCUs, so I'll just apply it always and "pour one out for the wasted bytes."

Reply:
I wrote the initial patch at #21396. The reason for the `ifdef` was to avoid the padding on platforms low on flash such as AVR. If wasting 1 to 3 bytes is not a big deal then the alignment can be unconditional.